### PR TITLE
Add coverage for the LID_USAGE section 

### PR DESCRIPTION
**Pull request by aerispaha**
### Summary 
This introduces coverage for the [LID_USAGE] inp section to the `swmmio.inp` API. 

Makes progress on #57

### CODE REVIEW

**Comment by aerispaha** [comment]
@asselapathirana I created this PR with your fork. However, your fork needs to be updated with the latest changes from aerispaha:swmmio, and merge conflict need to be resolved before we can move forward with this PR. I think the best way to do this is to include only the getter and setter methods for the `lid_usage` inp section, that you've proposed. 

Can you update your fork of swmmio accordingly? 

Thanks so much for contributing!

**Comment by asselapathirana** [comment]
Will update the code and get back. Thanks!

**Comment by asselapathirana** [comment]
Will do! :-)
Assela Pathirana,

File: swmmio/core.py [hunk 1]
```diff
@@ -611,10 +628,12 @@ def save(self, target_path=None):
         else:
             target_path = self.path
 
+
         for section in self._sections:
             # reformate the [SECTION] to section (and _section_df)
             sect_id = section.translate({ord(i): None for i in '[]'}).lower()
             sect_id_private = '_{}_df'.format(sect_id)
+            #print(sect_id_private)
```
Comment (aerispaha) [inline swmmio/core.py:636]: Can you remove this commented line?

File: swmmio/defs/inp_sections.yml [hunk 2]
```diff
@@ -63,7 +63,9 @@ inp_file_objects:
   XSECTIONS: [Link, Shape, Geom1, Geom2, Geom3, Geom4, Barrels, XX]
   POLLUTANTS: [Name, MassUnits, RainConcen, GWConcen, I&IConcen, DecayCoeff, SnowOnly,
                CoPollutName, CoPollutFraction, DWFConcen, InitConcen]
+
   INFLOWS: [Node, Constituent, Time Series, Type, Mfactor, Sfactor, Baseline, Pattern]
+  LID_USAGE: [Subcatchment,   'LID_Process',      Number,  Area,       Width,      InitSat,    FromImp,    ToPerv,     RptFile,                  DrainTo,          FromPerv,]
```
Comment (aerispaha) [inline swmmio/defs/inp_sections.yml:68]: Please replace this line with the following:
```yml
  LID_USAGE: [Subcatchment, LID_Process, Number, Area, Width, InitSat, FromImp, ToPerv, RptFile, DrainTo, FromPerv,]
```

This edit just removes the extra white space. Thanks!

File: swmmio/core.py [hunk 3]
```diff
@@ -857,6 +876,31 @@ def xsections(self, df):
...
+    def lid_usage(self):
+        """
+        Get/set LID_USAGE section of the INP file.
+        """
+        if self._lid_usage_df is None:
+            self._lid_usage_df = dataframe_from_inp(self.path, "[LID_USAGE]")
+        return self._lid_usage_df
+
+    @property
+    def raingages(self):
```
Comment (aerispaha) [inline swmmio/core.py:890]: This raingages getter section is already defined above around line 731. Please remove this property.

File: swmmio/core.py [hunk 4]
```diff
@@ -498,6 +499,20 @@ def headers(self):
 
         return self._rpt_section_details
 
+    @property
+    def external_outflow_volume(self):
```
Comment (aerispaha) [inline swmmio/core.py:503]: This is really cool! It is making think, though, that we need to provide coverage for the whole `Flow Routing Continuity` section of the rpt file. What do you think? 

If you agree, maybe we can tackle that in a separate issue. 
Then we could extend that logic to provide coverage for the rest of the rpt sections that follow this structure.

**Review by aerispaha — COMMENTED** [review]
Thanks for this PR, @asselapathirana! I have just a couple minor changes requested - with those resolved, we can get this into the next release.

File: swmmio/core.py (hunk 4 above)
Comment (aerispaha) [reply swmmio/core.py:503]: This could be handled in #194

**Review by asselapathirana — APPROVED** [review]

**Comment by aerispaha** [comment]
@asselapathirana I'm going to merge these changes, and then I'll address my comments in a separate PR

**Pull request merged**